# Bitcoin Statistical Inference

## Role
HAC Diebold–Mariano inference with Holm family-wise correction.

## Inputs
Ten frozen/reconstructed forecast vectors.

## Outputs
`bitcoin_dm_pairwise_results_hac_holm.csv`.

## Depends On
07–10.

## Authoritative Status
AUTHORITATIVE CORRECTED INFERENCE

## What This Notebook Does Not Do
It does not rerun models or silently replace the historical DM artifact.


In [1]:
from pathlib import Path
import sys
import pandas as pd
import numpy as np
ROOT = Path.cwd().parent if Path.cwd().name == 'notebooks' else Path.cwd()
sys.path.insert(0, str(ROOT))
from src.bitcoin_pipeline import *
RUN_GENERATION = False
PROMOTE_TO_AUTHORITATIVE = False


In [2]:
dm=pd.read_csv(ROOT/'results'/'bitcoin_dm_pairwise_results_hac_holm.csv'); assert dm.shape[0]==45 and len(set(dm['Model A'])|set(dm['Model B']))==10; dm

,Model A,Model B,N,HAC Lag,Mean loss differential,DM statistic,Raw p-value,Lower-RMSE model,Absolute RMSE difference,Relative RMSE difference,Practical interpretation,Holm-adjusted p-value,Significant raw?,Significant Holm?
0,Naive,Simple Exponential Smoothing — Rolling One-Step,1061,6,-7.814318e+03,-0.396730,6.915669e-01,Naive,2.106651,0.001137,Small,1.000000e+00,False,False
1,Naive,Additive-Trend Exponential Smoothing,1061,6,-6.734427e+04,-2.129474,3.321510e-02,Naive,18.077412,0.009752,Small,2.934753e-01,True,False
2,Naive,ARIMA Rolling One-Step,1061,6,-4.716156e+04,-1.145472,2.520136e-01,Naive,12.678085,0.006840,Small,1.000000e+00,False,False
3,Naive,7-Day Moving Average,1061,6,-5.561706e+06,-7.573065,3.645206e-14,Naive,1145.980299,0.618237,Moderate/Large,9.842055e-13,True,True
4,Naive,Prophet — 30-Day Periodic Refit,1061,6,-1.127975e+08,-8.787771,1.525571e-18,Naive,8927.538099,4.816260,Moderate/Large,5.918893e-17,True,True
5,Naive,Persistence-Enhanced Log-Return LSTM,1061,6,-1.025793e+05,-2.718383,6.560186e-03,Naive,27.466416,0.014818,Small,7.872224e-02,True,False
6,Naive,Persistence-Enhanced Log-Return Transformer,1061,6,-3.116394e+06,-12.305208,8.491742e-35,Naive,706.125036,0.380943,Small,3.736366e-33,True,True
7,Naive,Chronos-Bolt-Tiny,1061,6,-5.401428e+05,-5.294011,1.196622e-07,Naive,140.383153,0.075734,Small,2.632568e-06,True,True
8,Naive,TimesFM,1061,6,-2.666183e+05,-4.204846,2.612603e-05,Naive,70.574563,0.038074,Small,4.441425e-04,True,True
9,Simple Exponential Smoothing — Rolling One-Step,Additive-Trend Exponential Smoothing,1061,6,-5.952995e+04,-2.340088,1.927922e-02,Simple Exponential Smoothing — Rolling One-Step,15.970761,0.008606,Small,1.927922e-01,True,False


Squared-error loss is used. The Newey–West/Bartlett HAC lag is `floor(4*(N/100)^(2/9))` (6 at N=1,061). Raw and Holm-adjusted p-values are both preserved; Holm significance is the primary family-wise conclusion.